src/go/kreq-gen/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+set(GOPATH ${CMAKE_CURRENT_BINARY_DIR})
+
+function(add_go_dependency NAME MAIN_SRC)
```

Comment:
@ivotron @andrewhsu (all dev-prod) thoughts here? I abandoned the idea of using task to build this program and have it done before redpanda builds, because this presented with a weird cyclic dependency, i.e. i'd like to place this build product somewhere in `vbuild/<debug/release/clang` but it hasn't yet been created.

Also everything in the cmake project does not need to depend on anything in a folder outside of the builds context (vbuild/debug/clang), and if I were to build this via task, I'd need the CMake build system to know about `vbuild/go` something I thought was too against the grain.

On the other hand, this patch means that the `make` step will now depend on having internet access, something which currently isn't done now either. I thought about separating the go grab dependency step from the build step, placing the dependencies (franz-go) in `rp_build_deps` but go is really weird about GOPATH and where the projects build files and dependencies are located.

Feedback would be appreciated by anyone seeing this comment

Reply:
can `dev:install-go` be a dependency of `rp:build`? or why would that be a problem? it installs to `vbuild/go` and that can be added to `PATH` prior to invoking cmake. so basically assuming that go is already present by the time `kreq-gen/CMakeLists.txt` is executed

Reply:
This is what I plan to do to fix the CI failure, but it doesn't fix the aformentioned points. What you'd really need to do would be something along the lines of:
```
$ task dev:install-go # i think its ok to have rp:build depend on this, +1 to your idea about adding go to the PATH
$ task build-kafka-request-generator
$ mkdir vbuild/debug/clang/bin
$ cp vbuild/go/bin/kafka-request-generator vbuild/debug/clang/bin
$ task rp:build
```
And that seems odd to me

Reply:
For those interested, this commit to `vtools` [here](https://github.com/redpanda-data/vtools/commit/28cbaff08e817c6e7e2d069eb0d8758e135e0ae7`) has settled it.

The plan is to have redpandas build depend on an existing go installation and keep the source found here (invoking go through cmake)